### ISSUE TITLE
Sharding the second correction pass?

### ISSUE THREAD

**Issue by ASLeonard** [DecodeGenetics/Ratatosk#23]
Since the second correction pass takes significantly more time to finish, I was wondering if this can be split into multiple smaller jobs. It would be far less efficient, but assuming the `dbg.build`, `addCoverage`, and `detectSNPs` were executed using the full set of long reads, would it be possible then iterate over subsets of long reads in the [search algorithm](https://github.com/DecodeGenetics/Ratatosk/blob/74ca617afb20a7c24d73d20f2dcdf223db303496/src/Ratatosk.cpp#L371)? 

Basically it takes substantially longer to queue for the 120h partition (which is the max), and so is likely faster to run 10x jobs in the 24h partition.

If you think it may work, I'd be happy to give it a go and see what happens.

**Comment by ASLeonard** [comment]
I've been testing this out a bit more, and it sort of appears to work correctly. I was comparing the reads that got corrected before time ran out and reads that had been corrected after sharding, and they were generally the same.

After some more detailed investigating, I tried saving the dbg and graph data in "pass 2a", and then reading the dbg and graph data followed by the actual correction in "pass 2b". This also seemed to work fine and the corrections matched up.

I'm not sure how much might be changed in the next release, but it looks like the dbg gets built and written in pass 1. In pass 2, the dbg is built if it is pass 2 only, but read in otherwise (although the dbg should still be in memory in this case). Isn't it possible to also load the dbg if it is pass 2 only, it doesn't appear to have any different inputs?

**Comment by GuillaumeHolley** [comment]
Hi @ASLeonard,

Just to let you know that the next Ratatosk version will include an "index" subcommand to build the index required for one of the correction passes. The "correction" subcommand will either take the previously built index files or the usual raw input files. This will enable de novo correction over multiple machines.

**Comment by ASLeonard** [comment]
Great news. I hacked something like that together in the current release, but looking forward to the next version.
Thanks!
